src/server/game/Entities/Player/Player.h

```diff
@@ -1118,6 +1118,8 @@ class Player : public Unit, public GridObject<Player>
 
         static bool BuildEnumData(PreparedQueryResult result, WorldPacket* data);
 
+        bool IsImmunedToSpellEffect(SpellInfo const* spellInfo, uint32 index) const override; // override Unit::IsImmunedToSpellEffect
```

Comment:
and the Unit class has parent classes too. The comment explicitly states the overwritten method, which isn't a bad thing to have I suppose.